test/boost/sstable_set_test.cc

```diff
@@ -167,3 +173,137 @@ SEASTAR_TEST_CASE(test_partitioned_sstable_set_bytes_on_disk) {
         BOOST_REQUIRE_EQUAL(sst_set->bytes_on_disk(), ss1->bytes_on_disk() + ss2->bytes_on_disk());
     });
 }
+
+namespace dht {
+    dht::partition_range make_partition_range(const dht::token_range& tr) {
+        if (tr.is_full()) {
+            return dht::partition_range::make_open_ended_both_sides();
+        }
+        std::optional<dht::partition_range::bound> start_bound;
+        if (tr.start()) {
+            const auto& token = tr.start()->value();
+            start_bound = dht::ring_position::starting_at(tr.start()->is_inclusive() ? token : dht::next_token(token));
+            if (!tr.end()) {
+                return dht::partition_range::make_starting_with(*start_bound);
+            }
+        }
+        std::optional<dht::partition_range::bound> end_bound;
+        if (tr.end()) {
+            const auto& token = tr.end()->value();
+            end_bound = dht::ring_position::ending_at(tr.end()->is_inclusive() ? token : !token.is_last() ? dht::next_token(token) : dht::maximum_token());
+            if (!tr.start()) {
+                return dht::partition_range::make_ending_with(*end_bound);
+            }
+        }
```

Comment:
`nonwrapping_interval::transform()`

Reply:
I found a way to simplify the function using `transform_bound` instead.
Will include in next version